cmd/admin-profiling-stop.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * Minio Client (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/mc/pkg/probe"
+)
+
+var adminProfilingStopFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "save",
+		Usage: "The location of the zipped profiling files",
+	},
+}
+
+var adminProfilingStopCmd = cli.Command{
+	Name:            "stop",
+	Usage:           "Stop and download profiling data",
+	Action:          mainAdminProfilingStop,
+	Before:          setGlobalsFromContext,
+	Flags:           append(adminProfilingStopFlags, globalFlags...),
+	HideHelpCommand: true,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+    2. Download latest profiling data under the specified filename
+       $ {{.HelpName}} --save /tmp/profiling-data.zip myminio/
+
+`,
+}
+
+func checkAdminProfilingStopSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 1 {
+		cli.ShowCommandHelpAndExit(ctx, "stop", 1) // last argument is exit code
+	}
+
+	if len(ctx.String("save")) == 0 {
+		fatalIf(errDummy(), "Kindly specify the save path of profiling data using --save flag.")
+	}
+}
+
+// mainAdminProfilingStop - the entry function of profiling stop command
+func mainAdminProfilingStop(ctx *cli.Context) error {
+	// Check for command syntax
+	checkAdminProfilingStopSyntax(ctx)
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Create a new Minio Admin Client
+	client, err := newAdminClient(aliasedURL)
+	if err != nil {
+		fatalIf(err.Trace(aliasedURL), "Cannot initialize admin client.")
+		return nil
+	}
+
+	downloadPath := ctx.String("save")
+
+	// Add zip suffix if not found
+	if !strings.HasSuffix(downloadPath, ".zip") {
+		downloadPath += ".zip"
+	}
+
+	// Error out if target download file already exists
+	_, e := os.Stat(downloadPath)
```

Comment:
We can have a overwrite option as well... or move it automatically and save it with the timestamp? and write our anyways?

Reply:
> We can have a overwrite option as well... or move it automatically and save it with the timestamp? and write our anyways?

My taste prefers the current style, let's review with AB

Reply:
> We should move automatically and avoid this type of error reporting.. and just tell the user that this is what we did.

Done